scale_build/bootstrap/configure.py

```diff
@@ -0,0 +1,101 @@
+import os
+import shutil
+
+from scale_build.exceptions import CallError
+
+from .cache import check_basechroot_changed, create_basehash, save_build_cache, validate_basecache
+from .cleanup import remove_boostrap_directory
+from .logger import get_logger
+from .utils import BUILDER_DIR, CACHE_DIR, CHROOT_BASEDIR, get_apt_preferences, get_manifest, run
+
+
+def make_bootstrapdir(bootstrapdir_type, logger_file=None):
+    assert bootstrapdir_type in ('cdrom', 'package')
+    remove_boostrap_directory()
+    try:
+        _make_bootstrapdir_impl(bootstrapdir_type, logger_file)
+    finally:
+        remove_boostrap_directory()
+
+
+def _make_bootstrapdir_impl(bootstrapdir_type, logger_file=None):
+    logger = get_logger(bootstrapdir_type, 'w', logger_file)
+    run_args = {'logger': logger}
+    if bootstrapdir_type == 'cdrom':
+        deopts = ['--components=main,contrib,nonfree', '--variant=minbase', '--include=systemd-sysv,gnupg']
+    else:
+        deopts = []
+
+    # Check if we should invalidate the base cache
+    if validate_basecache(bootstrapdir_type):
+        logger.debug('Basechroot cache is intact and does not need to be changed')
+        return
+
+    run([
+        'apt-key', '--keyring', '/etc/apt/trusted.gpg.d/debian-archive-truenas-automatic.gpg', 'add',
+        os.path.join(BUILDER_DIR, 'keys/truenas.gpg')
+    ], exception=CallError, exception_msg='Failed adding truenas.gpg apt-key', **run_args)
+
+    apt_repos = get_manifest()['apt-repos']
+    run(
+        ['debootstrap'] + deopts + [
+            '--keyring', '/etc/apt/trusted.gpg.d/debian-archive-truenas-automatic.gpg', 'bullseye',
+            CHROOT_BASEDIR, apt_repos['url']
+        ], exception=CallError, exception_msg='Failed debootstrap', **run_args
+    )
+
+    create_basehash(bootstrapdir_type)
+    os.makedirs(os.path.join(CACHE_DIR, 'apt'), exist_ok=True)
+
+    run(['mount', 'proc', os.path.join(CHROOT_BASEDIR, 'proc'), '-t', 'proc'], **run_args)
```

Comment:
Do we need all this `os.path.join`? `f'{CHROOT_BASEDIR}/proc'` seems more compact